sql-plugin/src/main/scala/org/apache/spark/sql/rapids/collectionOperations.scala

```diff
@@ -581,8 +581,11 @@ case class GpuFloatArrayMin(child: Expression) extends GpuArrayMin(child) {
         // if all values in each list are nans or nulls
         val allNanOrNull = {
           val childIsNanOrNull = withResource(child.isNull()) {_.or(childIsNan)}
-          val nanOrNullList = withResource(childIsNanOrNull) {base.replaceListChild(_)}
-          withResource(nanOrNullList) {_.listReduce(listAll)}
+          withResource(childIsNanOrNull) { newChild =>
+            withResource(base.replaceListChild(newChild)) {
+              _.listReduce(listAll)
```

Comment:
I am confused on how this fixes it. Which one of these columns was closed too soon?

Reply:
This line `withResource(nanOrNullList) {_.listReduce(listAll)}` will discard the child of the new created list immediately. So in the next line, we access the list with child data discarded.

Reply:
Due to memory pooling, we still can read the child data after it was discarded and such data is still correct. That is why we didn't see any failed tests. However, if we disable partitioning and the input is large enough, the discarded (child) buffer got dirty immediately and leads to wrong output.

Reply:
ok looked at this with @jlowe and it is starting to make sense, but for the line above the one you quoted. `replaceListChild` conceptually isn't able to "inc ref count" `childIsNanOrNull` and so we need to make sure `childIsNanOrNull` is alive for the whole duration of this until `listReduce` is called, otherwise `base` has a closed child.

What is interesting is why this started failing now.

At least we should put a comment around this call explaining the life cycle, but ultimately we may want to change the API so it's safer.

Reply:
Yeah pooling could explain why it isn't seen often.